spec/features/post_index_spec.rb

```diff
@@ -0,0 +1,35 @@
+require 'rails_helper'
+
+RSpec.feature 'Post Index', type: :feature do
+  let(:user) { User.create(name: 'Tom', bio: 'He is a good programmer') }
+  let!(:post) { Post.create(author: user, title: "first post's title", text: 'first text') }
+  let!(:comment1) { Comment.create(author: user, post:, text: 'hello Tom') }
+  let!(:comment2) { Comment.create(author: user, post:, text: 'Thank You Tom') }
+  let!(:comment3) { Comment.create(author: user, post:, text: 'Good afternoon Tom') }
+  let!(:like1) { Like.create(user:, post:) }
+
+  before do
+    user.update(photo: 'https://www.kasandbox.org/programming-images/avatars/leaf-blue.png')
+  end
+
+  scenario "see user's profile picture, username, number of posts and interactions" do
+    visit user_posts_path(user)
+
+    expect(page).to have_selector('img[src="https://www.kasandbox.org/programming-images/avatars/leaf-blue.png"]')
+    expect(page).to have_content('Tom')
+    expect(page).to have_content('Number of posts: 1')
+    expect(page).to have_content('Comments: 3')
+    expect(page).to have_content('Likes: 1')
+  end
+
+  scenario 'see a section for pagination if there are more posts than fit on the view' do
+    visit user_posts_path(user)
+    expect(page).to have_selector('.pagination')
+  end
+
+  scenario "clicking on a post, it redirects me to that post's show page" do
+    visit user_posts_path(user)
+    click_link 'Post #1'
+    expect(page).to have_current_path(user_post_path(user, post))
```

Comment:
- [ ] You missed adding the below tests for the `posts#index` page, Please make sure to add them and they should pass without any errors.
  - Tests that assert you can see a post's title.
  - Tests that assert you can see some of the post's body.
  - Tests that assert you can see the first comments on a post.

Reply:
Resolved

Reply:
This change has been done well